### DISCUSSION TITLE
Permission Issue

### DISCUSSION THREAD

**Discussion by Anilkumar9010** [airbytehq/airbyte#49937] [Platform API Questions]
![image_2024_12_19T07_51_55_772Z](https://github.com/user-attachments/assets/776abe8d-3f84-4b0c-849a-2dd40552c55d)

I'm  Trying  to Connect the API's(like: v1/sources) to create a source ,destinations and connection from my local configuration . Unfortunately it is getting the permissions error . for your kind reference attached screen shot below .

**Comment by szemek** [comment]
https://reference.airbyte.com/reference/createsource
For Airbyte Cloud `POST /v1/sources`
For Airbyte OSS `/api/public/v1/sources`, requests should be sent to webapp

**Reply by Anilkumar9010** [reply]
Thank You For your kind information and it nis really valuable for me. And Successfully create a source and destination , now i'm trying to create connection between those but it throwing an error, so please help to resolve this .  and can u please verify the code i have written .
![Screenshot (113)](https://github.com/user-attachments/assets/f4b754ec-d218-47fe-b86a-c09c4b8134da)


![Screenshot (114)](https://github.com/user-attachments/assets/30999545-3e3e-41c9-829a-85117398ea75)

**Reply by szemek** [reply]
@Anilkumar9010 read documentation https://reference.airbyte.com/reference/createconnection
Probably everything you need is there .
`sourceId` and `destinationId` are required fields in body.
You can even generate sample code:
![image](https://github.com/user-attachments/assets/11c80e75-24da-49ac-9ec8-135c6e268de6)

**Reply by Anilkumar9010** [reply]
I  have follow the needed steps and i also follow using public keyword as you said for request. after sending this request, i am getting  403  type of response. so please help me to resolve this error is getting and i have attached the screenshot ca n u please check that issue and resolve the issue.

2)And I have given sourceId and DestinationId and also but still this error is occuring
![Screenshot (116)](https://github.com/user-attachments/assets/34a17721-30ec-4f48-a93a-f7b360a16791)

**Reply by Anilkumar9010** [reply]
If you are interest we can connect in any platform , Please help me to resolve this issue, and your reply valuable for me.

**Reply by szemek** [reply]
Please paste your PHP code

**Reply by Anilkumar9010** [reply]
Ok Sure Thank you.


I have given all the codes ,

1)Bearer token 
2)Used for Create a Source  
3)Used for create Destination 
4)Used for Connection 
5)Used for Sync the data. 

so can u please check the all methods in the pasted data and please resolve the issue .


[sourceanddestination.txt](https://github.com/user-attachments/files/18198157/sourceanddestination.txt)

**Reply by szemek** [reply]
I checked 2 methods `getBearerToken` and `createConnection`.
`getBearerToken` method works fine.
In `createConnection` I updated `sourceId`, `destinationId` and I used existing values from my local Airbyte instance. I also updated `authorization` header with value from `getBearerToken`. This method also works fine.

You need to check if source and destination work properly.
After creating them by API go to a source and press `Retest saved source`
Then the same thing for a destination, go to the destination and press `Retest saved destination`

**Reply by Anilkumar9010** [reply]
Thank you So much for giving time and resolving the issue Thank you.

**Comment by Skkaleshavali** [comment]
Hi @szemek  I need to create the Microsoft entra id source in our airbyte. I have use this OSS /api/public/v1/sources format and I also  add 
some inputs like  
/```
/getting airbyte token
    var token = await GetBearerToken();
if `(string.IsNullOrEmpty(token))`
    {
        return new JsonResult(new { error = "Failed to obtain access token" }) { StatusCode = 500 };
    }

    var client = new HttpClient();
    var requestBody = new
    {
        configuration = new
        {
            sourceType = "microsoft_entra_id",
            credentials = new
            {
                auth_type = "oauth2.0",
                client_id = clientId,
                client_secret = clientSecret,
                tenant_id = tenantId,
                user_id = userId
            }
        },
        name = "microsoft_entra_id_source",
        workspaceId = workspaceId
    };

    string jsonBody = JsonConvert.SerializeObject(requestBody);

    var request = new HttpRequestMessage
    {
        Method = HttpMethod.Post,
        RequestUri = new Uri($"{AirbyteApiUrl}/sources"), 
        Headers =
            {
                { "accept", "application/json" },
                { "authorization", $"Bearer {token}" },
            },
        Content = new StringContent(jsonBody, Encoding.UTF8, "application/json")
    };

    // Send request and check response
    using (var response = await client.SendAsync(request))
    {
        var body = await response.Content.ReadAsStringAsync();

        if (!response.IsSuccessStatusCode)
        {
            // Log the error response body for more details
            Console.WriteLine($"Error Response: {body}");
        }

        response.EnsureSuccessStatusCode();  // Ensure success
        var res = await response.Content.ReadFromJsonAsync<Models.Source>();
        return new JsonResult(res.SourceId);  // Return SourceId
    }
```
it's not working and this throws 400 bad request error

[a comment by an automated account is not shown]
